tools/mayan_sync_helper/mayan_sync_transformer.py

```
@@ -0,0 +1,220 @@
+
+        if project == 'x':
+            categories.append('PROJECT')
+        if research == 'x':
+            categories.append('RESEARCH')
+        if acquisition == 'x':
+            categories.append('ACQUIRE')
+        if lease == 'x':
+            categories.append('LEASLIC')
+        if disposition == 'x':
+            categories.append('DISPOSE')
+        if management == 'x':
+            categories.append('MANAGEMENT')
+
+        document_types.append(DocumentType(
+            doc_type_code, doc_type_name, doc_type_purpose, display_order, categories))
+
+    # Order by label
+    document_types.sort(key=lambda x: x.label, reverse=False)
+    for i in range(0, len(document_types)):
+        document_types[i].display_order = i
+
+    return document_types
+
+
+def load_metadata_definitions():
+    # Give the location of the file
+    metadata_file_location = ("xml_definitions/metadata_definitions.xlsx")
+
+    # To open Workbook
+    wb = xlrd.open_workbook(metadata_file_location)
+    metadata_sheet = wb.sheet_by_index(0)
+
+    metadata_types = []
+
+    # Get the document Type
+    for i in range(3, metadata_sheet.nrows):
+        metadata_code = metadata_sheet.cell_value(i, 1)
+        metadata_label = metadata_sheet.cell_value(i, 2)
+
+        print(metadata_code, metadata_label)
+
+        metadata_types.append(MetadataType(metadata_code, metadata_label))
+
+    return metadata_types
+
+
+def match_doc_metadata(document_types, metadata_types):
+    # Give the location of the file
+    loc_match = ("xml_definitions/doc_metadata.xlsx")
+
+    # To open Workbook
+    wb = xlrd.open_workbook(loc_match)
+    sheet = wb.sheet_by_index(0)
+
+    doctype_sort_order = []
+
+    for i in range(3, sheet.nrows):
+        print("--------------" + str(i))
+        current_row_name = sheet.cell_value(i, 1)
+
+        if current_row_name != "":
+            doc_type_name = current_row_name
+
+        if doc_type_name not in doctype_sort_order:
```

Hmm, we may want to throw here instead, I feel like I'd rather fail fast then accidentally generate a json with missing data.